waiter/integration/waiter/token_request_test.clj

```diff
@@ -679,6 +680,79 @@
       (delete-token-and-assert waiter-url token)
       (delete-service waiter-url service-id))))
 
+(defmacro run-token-param-support
+  [waiter-url created-service-ids request-fn in-headers expected-env]
+  `(let [request-headers# ~in-headers
+         response# (make-request-with-debug-info request-headers# ~request-fn)
+         service-description# (response->service-description ~waiter-url response#)
+         service-id# (:service-id response#)]
+     (is (= 200 (:status response#)))
+     (is (= ~expected-env (:env service-description#)) (str service-description#))
+     (swap! ~created-service-ids conj service-id#)
```

Comment:
I agree with @sradack , it would be nice to refactor to not need `created-service-ids`